### PR TITLE
Moved the CZ URL and node registration logic of the driver to be more…

### PR DESCRIPTION
**Pull request by shamsasari**
… internal, not available through the standard driver call, as these are not testing features for an app dev.

Also cleanup up some of the related tests.

### CODE REVIEW

File: node/src/integration-test/kotlin/net/corda/node/services/network/NetworkMapClientTest.kt [hunk 1]
```diff
@@ -4,85 +4,96 @@ import net.corda.core.node.NodeInfo
...
     private val portAllocation = PortAllocation.Incremental(10000)
 
+    private lateinit var networkMapServer: NetworkMapServer
+    private lateinit var compatibilityZone: CompatibilityZoneParams
+
+    @Before
+    fun start() {
+        networkMapServer = NetworkMapServer(cacheTimeout, portAllocation.nextHostAndPort())
+        val address = networkMapServer.start()
+        compatibilityZone = CompatibilityZoneParams(URL("http://$address"), rootCert = null)
```
Comment (al-r3) [inline node/src/integration-test/kotlin/net/corda/node/services/network/NetworkMapClientTest.kt]: Use the 4 params constructor of URL

File: node/src/integration-test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperDriverTest.kt [hunk 2]
```diff
@@ -30,153 +29,108 @@ import java.net.URL
...
+        internalDriver(
+                portAllocation = portAllocation,
+                notarySpecs = emptyList(),
+                compatibilityZone = compatibilityZone
         ) {
-            assertThatThrownBy {
-                startNode(providedName = ALICE_NAME, initialRegistration = true).get()
-            }.isInstanceOf(java.nio.file.NoSuchFileException::class.java)
+            startNode(providedName = CordaX500Name("Alice", "London", "GB")).getOrThrow()
+            assertThat(registrationHandler.requests).containsKeys("Alice")
```
Comment (al-r3) [inline node/src/integration-test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperDriverTest.kt]: Strictly speaking this is more lax than the previous implementation.
This doesn't verify that the node requests for the certificate approval status.

File: node/src/test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperTest.kt [hunk 3]
```diff
@@ -92,4 +91,51 @@ class NetworkRegistrationHelperTest {
...
+        val certService = rigorousMock<NetworkRegistrationService>()
+
+        assertThatThrownBy {
+            NetworkRegistrationHelper(config, certService)
+        }.hasMessageContaining(config.rootCertFile.toString())
+    }
+
+    @Test
+    fun `root cert in response doesn't match expected`() {
+        val config = testNodeConfiguration(
```
Comment (al-r3) [inline node/src/test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperTest.kt]: config and identities can be moved up to the class level. Now they're duplicate between tests.

File: node/src/integration-test/kotlin/net/corda/node/services/network/NetworkMapClientTest.kt (hunk 1 above)
Comment (shamsasari) [reply node/src/integration-test/kotlin/net/corda/node/services/network/NetworkMapClientTest.kt]: This is easier to use and clearer. The 4th param in the 4 param c'tor ends up being an empty string which isn't obviously clear why.

File: node/src/integration-test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperDriverTest.kt (hunk 2 above)
Comment (shamsasari) [reply node/src/integration-test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperDriverTest.kt]: True. I'll update the registration handler.

File: node/src/test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperTest.kt (hunk 3 above)
Comment (shamsasari) [reply node/src/test/kotlin/net/corda/node/utilities/registration/NetworkRegistrationHelperTest.kt]: Yes

**Comment by shamsasari** [comment]
Comments addressed.

**Pull request merged**